journals/week3.md

```diff
@@ -0,0 +1,189 @@
+# Week 3: Exploration
+### Finding the Appropriate Type for Flux Msgcounters
+
+Last week I began testing on some "real" dummy data using the builtin message counters attached to each flux handle to get stats on each module. Originally I tried sending them as meters on accident as brubeck uses the `c` for meters while statsd uses it for counters, so then I did most testing sending the stats as gauges (`g`) which turned out to not be the most useful representation as the values were just increasing. This time around, I sent them as counters (`C`), and the resulting graphs are more useful as they show the change per flush interval; in other words, for each message type, the count that was received within the flush interval is what's sent along to Graphite. In addition to using a more useful metric type, I also used a different and possibly more practical watcher within the flux broker. I used a `flux_timer_watcher` with a `1s` wake up time to send the metrics once per second instead of being based on the blocking/unblocking/idling of the reactor loop. The timer callback function is identical to the one used within the check and idle watchers from last time. After the changes were made, it was time to test it out by running lots of simple jobs.
+
+```bash
+# enable fripp debugging on all modules
+$ for m in $(flux module list | awk 'FNR !~ /^1$/ {print $1}'); do flux module debug -s 0x1000 $m; done
+
+# run 10,000 simple jobs one at a time
+$ for i in {1..10000}; do flux mini run echo -n "$i " && hostname; done
+```
+
+**took ~ 1h1m44s**
+
+job-ingest message counters received per 10s flush interval
+
+![job-ingest msg counters](https://i.imgur.com/qrhu9r5.png)
```

Comment:
I failed to notice this the last couple weeks, but I wonder if it would be better to capture all images into the repo instead of using a 3rd party hosting service?